src/neo/SmartContract/ApplicationEngine.Runtime.cs

```diff
@@ -354,5 +360,25 @@ protected internal void BurnGas(long gas)
                 throw new InvalidOperationException("GAS must be positive.");
             AddGas(gas);
         }
+
+        /// <summary>
+        /// Attribute to defend against Re-entry attack
+        /// </summary>
+        /// <param name="method"></param>
+        /// <param name="paramCount"></param>
+        protected internal void NoReentry(string method, int paramCount)
+        {
+            if (_noReentryMethods.ContainsKey(CurrentScriptHash))
+            {
+                if (_noReentryMethods[CurrentScriptHash].Contains(method + paramCount))
+                    throw new InvalidOperationException(
+                        $"Cannot Call Method {method} Of Contract {CurrentScriptHash} in an reentry manner.");
+            }
+            else
+            {
+                _noReentryMethods[CurrentScriptHash] = new List<string>();
+            }
+            _noReentryMethods[CurrentScriptHash].Add(method + paramCount);
+        }
```

Comment:
What if a method called 'Hello1' and parameter is count is '0' while another method is 'Hello' and parameter count is '10'?

Reply:
@dusmart  addressed